### ISSUE TITLE
Team up with someone willing to sign the drivers

### ISSUE THREAD

**Issue by Redsandro** [t-d-k/LibreCrypt#6]
After I [wondered](https://en.wikipedia.org/wiki/Talk:FreeOTFE#Anyone_know_if_the_author_is_still_alive.3F) if the original author of `FreeOTFE` had disappeared when I couldn't get in touch, I started using a virtual machine running Ubuntu to mount my `LUKS` drive as a virtual share in Windows.

This is tedious, and I accidentally stumbled upon ~~`DoxBox`~~ `LibreCrypt` today, so I thought I'd bring back my quest to get the ~~`FreeOTFE`~~ ~~`DoxBox`~~ `LibreCrypt` drivers signed.

What I was originally trying to suggest to _Sarah Dean_ (`FreeOTFE` author) is to ask these guys to review the code and help with driver signing: http://reactos.org/wiki/Driver_Signing

> The ReactOS Foundation has a VeriSign code signing certificate that can be used to sign 64bit drivers for Windows. The Foundation is prepared to assist other open source projects that wish to distribute 64bit drivers but are not able to acquire a code signing certificate by signing their drivers. However, this offer is contingent on fulfilling the following conditions.
> - The Foundation will only sign open source drivers.
> - ...

If stable releases could be signed, the software would be more convenient and easy to use, and gain popularity.

**Comment by t-d-k** [comment]
Thanks for suggesting this. It is a good idea, but I want to wait till the drivers are stable before asking them to sign the drivers, otherwise they would have to sign them again.
I plan to update the drivers at some point with the latest Gladman library, so after that would be a good time.

**Comment by Redsandro** [comment]
I hope it works out. This would be sweet.

**Comment by linux-modder** [comment]
Any update on the drivers @t-d-k ?

**Comment by t-d-k** [comment]
I've contacted @reactos, but got no reply. The other option is to fund-raise it from users and supporters. 
It's not a lot of money, but I don't want to have to pay it myself and any fund-raising should be transparent, so I'm looking into ways of doing this ATM.

**Comment by jdevora** [comment]
Hi,

I'm willing to participate, on what range of money are we talking about?
Properly marketed as a True-crypt alternative, I think it could attract 
some people to  a crowd funding initiative.

Cheers
  JD

**Comment by Redsandro** [comment]
`FreeOTFE` had two audiences, and you have to make sure `LibreCrypt` markets them both.

1. TrueCrypt alternative
1. Windows driver for mounting `LUKS` drives.

Usually, the one audience doesn't care about the other functionality. Personally, I just want to mount `LUKS` drives in Windows (and hope `EcryptFS` will be supported too).

**Comment by Redsandro** [comment]
@t-d-k said:
> I've contacted @reactos, but got no reply.

@jdevora said:
> I'm willing to participate

@reactos didn't commit for a while but they seem active on Twitter. I left a [tweet](https://twitter.com/Redsandro/status/605358965434793985). If people [retweet](https://twitter.com/intent/retweet?tweet_id=605358965434793985) this, perhaps it will get noticed. It would be sweet if @reactos was to respond here. :smile:

**Comment by linux-modder** [comment]
@Redsandro, what is the push for ecryptfs from?

**Comment by t-d-k** [comment]
@jdevora 
It's [$178](https://www.digicert.com/code-signing/kernel-mode-certificates.htm).
But people have also been asking for a code review - this would be more costly if paid for. So I'd rather have something open-ended than a Pledgie, like just putting a paypal link up.

@Redsandro 
From the [survey](https://www.surveymonkey.com/results/SM-LBR8RBYD/) and issues raised it seems most people are using it for LUKS.
I hope to add a feature to create LUKS containers and make this the default type, which would make it more secure as well as improve Linux compatibility.
Thanks for the tweet, I've retweeted it.

**Comment by Redsandro** [comment]
@linux-modder said:
> @Redsandro, what is the push for ecryptfs from?

While `LUKS` is the de-facto disk-based-encryption method in Linux, `EcryptFS` is the default file-based-encryption method in most linux distro's. I use both. I would love to not only mount my `LUKS` drives in Windows, but also my `EcryptFS` home directory on non-LUKS drives, or my `EcryptFS` share on Dropbox.

**Comment by jdevora** [comment]
@linux-modder:
ecryptfs encrypts at a file level, that makes it ideal for use it for  cloud based storage, when you change a file you don't need to synchronize the whole container, only the encrypted version of the file

**Comment by linux-modder** [comment]
I am familiar with ecryptfs just not seeing the reason for extra complexity on windows side if not using a ntfs /fat fs

**Comment by Redsandro** [comment]
@linux-modder Indeed, there is no reason to bake a cake if you are not going to eat it. After all, you could say the exact same thing for `LUKS` (_Linux_ Unified Key Setup) usage, one of the main features of `LibreCrypt`. It is file-system independent, it just exposes whatever filesystem you choose, and by default Linux sets up an `ext` filesystem.

Those of us that have to work with both Windows and Linux usually use `NTFS` so we can mount our drives in both Windows and Linux.

I use my `LUKS` drives in both Windows and Linux using `LibreCrypt`. However, for `ecryptfs` encrypted parts of my non-`LUKS` `NTFS` drives , I have to resort to clumsy Vagrant solutions. @jdevora and I already mentioned cloud storage like Dropbox where a filesystem is irrelevant.

I actually started to 'port' my `LUKS` drives to normal drives with `ecryptfs` sections for multiple reasons:

* We can encrypt only the sensitive data, leaving the rest accessable, speeding up the read/write process on lightweight computers
* Use the full available free space of external drives on others' computers, not having to split the space between a `LUKS`- and non-`LUKS` partition.
* Easily sync private projects to the cloud

The _only_ reason I'm still having some `LUKS` partitions is because the Vagrant solution is tedious for sharing my files between Windows and Linux.

Hence, there is good reason, and you can see it. You are just not personally interested.

**Comment by alvinhochun** [comment]
At least one member acknowledged this and also responded to the above tweet.

Perhaps you guys should have just hoped into freenode \#reactos and ask directly :P

(Note: I am not an official ReactOS developer)

**Comment by Redsandro** [comment]
@alvinhochun great news indeed!

@reactos said 4 hours ago:
> Hi @Redsandro @LibreCrypt ! Sorry the delay! I've pushed up your driver signing request. I'll answer back asap. :)

I hope this works out and we'll have `LibreCrypt` working without enabling Windows testmode soon. :)

**Comment by bragin** [comment]
Yes, we would need to conduct a code review and then we'd be glad to sign your driver.

**Comment by linux-modder** [comment]
@reactos team,   If you need help I can be reached here or email
sheldon.corey@gmail.com  -- would assume same for tdk but can't speak for
him..

Corey W Sheldon
Freelance IT Consultant, Multi-Discipline Tutor
(p) 310.909.7672
G+: <https://www.plus.google.com/+CoreySheldon>
LinkedIn: <https://www.linkedin.com/profile/view?id=70127804>
Github: <https://www.github.com/linux-modder>
Facebook: <https://www.facebook.com/corey.sheldon>
Several Communities on Stack Exchange <https://www.stackexchange.com>

<http://www.facebook.com/1stclassmobileshine>

Tutoring in person or via any of the following platforms:
HackHands <https://www.hackhands.com>
Wizpert <https://www.wizperts.com>
FieldNation <https://www.fieldnation.com>
AirPair <https://www.airpair.com>
Truelancer <http://www.truelancer.com>

{PayPal,Google Wallet/Play store, Apple Pay}
---------------------------------------------------------------------------------------------------
pub  3072D/718BF597
<http://pgp.mit.edu/pks/lookup?op=get&search=0xE958C5D6718BF597> 2014-12-08
      Key fingerprint = 2930 99EB 083D D332 0752 88C4 E958 C5D6 718B F597
uid     Corey Sheldon (Fedora Key) <sheldon.corey@gmail.com>
---------------------------------------------------------------------------------------------------

**Comment by Redsandro** [comment]
I hope @t-d-k will be back soon and negotiate how to make this as easy as possible for @bragin.

**Comment by t-d-k** [comment]
@bragin  
This is wonderful news. Thank you very much.

[Technical Details: Building the Software](https://github.com/t-d-k/LibreCrypt/blob/master/docs/technical_details__build_notes.md#building_kernel_drivers) has details on how to build the drivers.
If you have any questions please contact me on tdk at squte.com, or here, or on the forums at http://doxbox.squte.com/groups/doxbox
tdk

**Comment by AndreasA** [comment]
Any news regarding signed drivers?
Because I would really like to be easily share LUKS encrypted drives between Windows and Linux on 64Bit.

**Comment by Manouchehri** [comment]
@bragin @reactos Any status update on the code review?

**Comment by Manouchehri** [comment]
With #38 and #39 still open, I'm going to guess that's put a halt on getting the driver signed?

**Comment by t-d-k** [comment]
I haven't heard anything more from @bragin or @reactos. 
This will be the first thing done when the donations are enough.

**Comment by Manouchehri** [comment]
If #38 and #39 are fixed, I may be able to provide signing with my certificate.

**Comment by AndreasA** [comment]
I am also hoping for signed drivers soonish :)
Btw. does LibreCrypt also work with Windows 10?

**Comment by linux-modder** [comment]
Andreas,


Been doing some drive-by testing around work schedule  on Win 10 Pro 64
(efi +sb in play) and  it  seems to work on usbs  not  hard drive
containers (using lvm) tho (I use LVM on my  crypts)

Corey W Sheldon
Freelance IT Consultant, Multi-Discipline Tutor
Ameridea LLC, Founder, CTO
http://github.com/ameridea
Fedora Ambassador, North America
http://getfedora.org
Server Intern Staff, Citygate
http://citygate.org
(p) +1 (310) 909-7672
Find Me  on any of the sites  I teach /frequent:
https://gist.github.com/linux-modder/ac5dc6fa211315c633c9
----------------------------------------------------------------------------------------------------------------------------------------------
    "Have no way as way, no limitation as limitation. One must never
underestimate the power of boredom...from which creativity and laziness are
borne, which can spark great works of chaos and genius."
----------------------------------------------------------------------------------------------------------------------------------------------
PGP: 0xe958c5d6718bf597 FP = 2930 99EB 083D D332 0752 88C4 E958 C5D6 718B
F597 linuxmodder@keybase.io
Tox: Corey84 || Linux-modder
9357BC6A5944A08AFC7D1EFFD61F6A73B9EABF8B2FB84ACF1DAC9A1A4D0A4705FFCCD0E5499B
Linphone: sip:linuxmodder
BitAddress:15cn1BvAFEREHk8UekJ6i9Dxi9Wbw6vzDD
